azuredata/src/main/java/com/azure/data/service/Response.kt

```diff
@@ -58,4 +62,22 @@ open class Response<T>(
      * Returns the associated value of the result if it is a success, null otherwise.
      */
     val resource: T? = result.resource
+
+    /**
+     * Returns `true` if there are more paged results available
+     */
+    val hasMoreResults : Boolean get() {
+        return !metadata.continuation.isNullOrEmpty()
+    }
+
+    fun <T : Resource> next(callback: (ListResponse<T>) -> Unit) {
+        val documentClient = documentClient
+                ?: return callback(ListResponse(DataError(DocumentClientError.NextCalledTooEarlyError)))
+
+        val listResponse : ListResponse<T> = this as ListResponse<T>
+
+        val documentClass: Class<T> = documentClass as Class<T>
+
+        documentClient.next(listResponse, documentClass, callback)
+    }
```

Comment:
Let's see if this is possible to implement as an extension to `ListResponse`... not sure if that works on a typealias, but would love for the API to only show up on things where it makes sense (queries, lists)